### DISCUSSION TITLE
Embed signed hash into PDF

### DISCUSSION THREAD

**Discussion by fedegalan** [MatthiasValvekens/pyHanko#103] [Q&A]
Hello, I have a hash signed with an external website, the hash is in pkc7s format, how do I integrate it back into the PDF?

**Comment by MatthiasValvekens** [comment]
Hi,

Sorry for the late response, I was attending a conference last week. 

It sounds like you're dealing with an external signature provider that supplies complete CMS objects (CMS = the successor standard to PKCS#7). For the record, that's not quite the same use case as #90: there, it turned out that the server only supplied raw signed hashes.

Do you have access to the certificates ahead of time (i.e. before initiating the singing process)?

If yes: have you read [the documentation on interrupted signing](https://pyhanko.readthedocs.io/en/latest/lib-guide/signing.html#interrupted-signing)?

If you don't have access to the certs, then you'll have to look at the [PdfCMSEmbedder docs](https://pyhanko.readthedocs.io/en/latest/lib-guide/signing.html#the-low-level-pdfcmsembedder-api).
